### PR TITLE
Obs rsync list of projects

### PR DESCRIPTION
**Pull request by andrii-suse**
* Let configure OBS repositories per project;
* Use configured OBS repository in API request for checking dirty status; 
* Add api route to return json with all configured OBS (project, repository) pairs;
* Add optional repository parameter to to sync call to properly react on which repository of OBS project must be synced.

### CODE REVIEW

File: lib/OpenQA/WebAPI/Plugin/ObsRsync/Controller/Folders.pm [hunk 1]
```diff
@@ -26,21 +41,25 @@ sub index {
...
     }
 
     for my $folder (@$folders) {
         my $alias = $folder;
         $alias = "$obs_project|$folder" if $obs_project;
         my $run_last_info = $helper->get_run_last_info($alias);
         my ($fail_last_job_id, $fail_last_when) = $helper->get_fail_last_info($alias);
+        my $dirty_status = $helper->get_dirty_status($obs_project // $folder);
+        my $api_repo     = $helper->get_api_repo($obs_project     // $folder);
+        $dirty_status = "$dirty_status ($api_repo)" if $api_repo ne "images";
```
Comment (kalikiana) [inline lib/OpenQA/WebAPI/Plugin/ObsRsync/Controller/Folders.pm]: Why is `images` excempt here? :thinking:

File: t/api/14-plugin_obs_rsync.t [hunk 2]
```diff
@@ -60,6 +60,10 @@ my $app = $t->app;
 $t->ua(OpenQA::Client->new(apikey => 'ARTHURKEY01', apisecret => 'EXCALIBUR')->ioloop(Mojo::IOLoop->singleton));
 $t->app($app);
 
+# just check that all prjects are mentioned
```
Comment (kalikiana) [inline t/api/14-plugin_obs_rsync.t]:
```suggestion
# just check that all projects are mentioned
```

File: lib/OpenQA/WebAPI/Plugin/ObsRsync.pm [hunk 3]
```diff
@@ -261,6 +264,42 @@ sub _get_dirty_status {
...
+
+# Build url to check dirty status for project
+sub _get_api_dirty_status_url {
+    my ($c, $project) = @_;
+    my $helper = $c->obs_rsync;
+    my $url    = $helper->project_status_url;
+    return "" unless $url;
+    $url =~ s/%%PROJECT/$project/g;
+    my $package = $helper->get_api_package($project);
+    $url = "$url?package=$package" if $package;
```
Comment (kalikiana) [inline lib/OpenQA/WebAPI/Plugin/ObsRsync.pm]:
```suggestion
    $url->query({package => $package}) if $package;
```

File: lib/OpenQA/WebAPI/Plugin/ObsRsync/Controller/Folders.pm (hunk 1 above)
Comment (andrii-suse) [reply lib/OpenQA/WebAPI/Plugin/ObsRsync/Controller/Folders.pm]: `images` is default value, because it is used in most of projects.

**Comment by andrii-suse** [comment]
> I assume you do some sort of manual integration tests with the OBS-side?

Actually current unit tests properly cover most of functionality, and coupling `openqa-trigger-from-obs` is very small, so nothing should be different than in unit tests. Some parts are in development in parallel (like passing new repository parameter), and are needed to address current issues.

File: lib/OpenQA/WebAPI/Plugin/ObsRsync.pm (hunk 3 above)
Comment (andrii-suse) [reply lib/OpenQA/WebAPI/Plugin/ObsRsync.pm]: Url is a string, so it doesn't work this way

Comment (Martchus) [reply lib/OpenQA/WebAPI/Plugin/ObsRsync.pm]: Yes, the suggested code is wrong. But the point is that you should make the URL with `Mojo::URL` so parameters are correctly encoded.

**Pull request merged**